shared/common-adapters/list-item2.tsx

```diff
@@ -21,30 +21,37 @@ type Props = {
   body: React.ReactNode
   firstItem: boolean
   action?: React.ReactNode
-  onlyShowActionOnHover?: 'appear' | 'animate' | null
+  onlyShowActionOnHover?: 'fade' | 'grow' | null
   onClick?: () => void
 }
 
 const HoverBox = Styles.isMobile
   ? Box2
   : Styles.styled(Box2)({
       // @ts-ignore
-      '.hidden-no-hover': {
-        justifyContent: 'flex-end',
+      '.fade': {
+        opacity: 0,
+        ...Styles.transition('opacity'),
+      },
+      '.grow': {
         maxWidth: 0,
         overflow: 'hidden',
-      },
-      '.hidden-no-hover-animate': {
         ...Styles.transition('max-width'),
       },
       ':hover': {
         backgroundColor: Styles.globalColors.blueLighter2,
       },
       ':hover .avatar-border': {
+        // TODO: it'd be nice to move this out of this file since list-item2
+        // has nothing todo with avatars. We'd need Kb.AvatarLine to know
+        // about different background and set that on hover though.
         boxShadow: `0px 0px 0px 2px ${Styles.globalColors.blueLighter2} !important`,
       },
-      ':hover .hidden-no-hover': {
-        maxWidth: 64,
+      ':hover .fade': {
+        opacity: 1,
```

Comment:
This is where it went wrong for the editing row. The `max-width` is limiting the width of actions for all rows. So instead, don't use max-width for the "appear" case, and use `opacity` instead.

Also instead of a fixed `max-width` in the `:hover` case, use `'unset'`.